test-workflows/task/eks-cluster-create/eksctl/0.1/create-eks-cluster.yaml

```diff
@@ -0,0 +1,65 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: eks-cluster-create
+  namespace: hakuna
+spec:
+  description: |
+    Create an EKS cluster.
+    This Task can be used to create an EKS cluster in an AWS account and write a kubeconfig to a desired location that
+    can be used by other tasks (in a context with kubectl) to make requests to the cluster.
+  params:
+  - name: cluster-name
+    description: The name of the EKS cluster you want to spin.
+  - name: eks-version
+    default: "1.20"
+    description: The EKS version to install.
+  - name: region
+    default: us-west-2
+    description: The region where the cluster is in.
+  - name: zones
+    default: us-west-2a,us-west-2b,us-west-2c
+    description: The zones where the cluster is in.
+  - name: nodegroup-name
+    default: linux-nodes
+    description: The name of the nodegroup of the cluster.
+  - name: node-type
+    default: m5.xlarge
+    description: The type of the EC2 instaces for the nodegroup of the cluster.
+  - name: desired-nodes
+    default: "10"
+    description: The desired number of nodes in the cluster.
+  - name: min-nodes
+    default: "1"
+    description: The minimum number of nodes in the cluster.
+  - name: max-nodes
+    default: "10"
+    description: The maximum number of nodes in the cluster.
+  workspaces:
+  - name: secrets
+    mountPath: /tekton/home/.aws
+    description: The secret with the AWS keys
+  - name: config
+    description: |
+      A workspace into which a kubeconfig file called `kubeconfig` will be written that will contain the information required to access the cluster. The `kubeconfig` will expect to use [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/) to authenticate, so in order for it to be used it must be run in a container which contains both `kubectl` and `aws-iam-authenticator`.
+  steps:
+  - name: write-kubeconfig
+    image: docker.io/weaveworks/eksctl:0.35.0@sha256:48c1fa508970a01fd87a73ac7932a7160479d678cd019a3c84533d911fc54327
+    script: |
+      echo "Starting to create eks cluster"
+      eksctl create cluster \
+        --name $(params.cluster-name) \
+        --version $(params.eks-version) \
+        --region $(params.region) \
+        --zones $(params.zones) \
+        --nodegroup-name $(params.nodegroup-name) \
+        --node-type $(params.node-type) \
+        --nodes $(params.desired-nodes) \
+        --nodes-min $(params.min-nodes) \
+        --nodes-max $(params.max-nodes)
+      if [ $? -eq 0 ]
```

Comment:
You can try setting the bash flag `-e` which will exit on any command's error. 

`#!/usr/bin/env bash -e`